scripts/translate_pdf.py

```
@@ -0,0 +1,725 @@
+                            
+                            # Check for result mismatch
+                            if len(batch_results) != len(batch_items):
+                                logger.warning(f"Batch {batch_idx} result count mismatch ({len(batch_results)} vs {len(batch_items)}). Using fallbacks.")
+                                # Pad with original text or truncate
+                                if len(batch_results) < len(batch_items):
+                                    batch_results.extend(batch_items[len(batch_results):])
+                                else:
+                                    batch_results = batch_results[:len(batch_items)]
+                            
+                            # Add to translations dict and cache
+                            for text, translation in zip(batch_items, batch_results):
```

Ruff (0.14.5)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)